### PR TITLE
.into<Library>()

### PR DESCRIPTION
**Pull request by JasonCubed132**
Modifies dcspkg to provide almost all of its internal structures and functions as a library.

### CODE REVIEW

**Comment by Joeyh021** [comment]
Looks good on a code level to me. Few comments w.r.t. api/library design though:

- The modules shouldn't be `pub mod`, we don't want to export everything in the crate
- I don't like the idea of exposing the `DcspkgConfig` struct in the public API. `lib.rs` should `pub use` any functions from `commands` that it wishes to re-export.
  - To this end, `commands::run` should probably be re-written to take the parameters separately instead of the struct
  - `cli.rs` should be part of the binary crate, not the lib
  - Your public API should be documented with doc comments. Run `cargo doc --open` to browse the API and see what's exported.

**Comment by Joeyh021** [comment]
Oh, and run `cargo fmt`, not sure why the CI hasn't run 🤔

**Comment by Joeyh021** [comment]
Changes look good.
Instead of wrapping private functions with public ones in `lib.rs`, you can `pub use` the ones from the private modules to export individual items publicly. If you make that change I'm happy to merge :)

**Comment by JasonCubed132** [comment]
Side note: I cannot figure out how to fix the unused code warnings. If I make them un-public, other files can't find them.

**Comment by Joeyh021** [comment]
> Side note: I cannot figure out how to fix the unused code warnings. If I make them un-public, other files can't find them.

That's because `util` is a subodule of both the bin and lib crate. Make it a submod of the the lib and then import `dcspkg::util` in the bin crate root.

File: dcspkg/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,12 @@
 use serde::{Deserialize, Serialize};
 
+mod commands;
+pub mod config;
+pub mod util;
+
+pub use crate::commands::{list_all_packages, install_package, run_package};
+pub use crate::util::list_installed_packages;
```
Comment (Joeyh021) [inline dcspkg/src/lib.rs]: You can remove this line: the function is exported anyway as util is a `pub mod`: see the site generated by `cargo doc` and it's there twice.

**Comment by Joeyh021** [comment]
Fantastic, tysm for this

**Pull request merged**